### PR TITLE
QF-918 Upgrade to Azure Identity - Rework

### PR DESCRIPTION
**Pull request by SawNawDfE**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeFeedback.Data/ApprenticeFeedbackDataContext.cs [hunk 1]
```diff
@@ -30,7 +31,7 @@ public class ApprenticeFeedbackDataContext : DbContext,
     {
         private const string AzureResource = "https://database.windows.net/";
         private readonly ApplicationSettings _configuration;
-        private readonly AzureServiceTokenProvider _azureServiceTokenProvider;
+        private readonly ChainedTokenCredential _azureServiceTokenProvider;
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ApprenticeFeedback.Data/ApprenticeFeedbackDataContext.cs:34]: Should rename the variable here

File: src/SFA.DAS.ApprenticeFeedback.Data/ApprenticeFeedbackDataContext.cs [hunk 2]
```diff
@@ -59,7 +60,7 @@ public ApprenticeFeedbackDataContext(DbContextOptions<ApprenticeFeedbackDataCont
         {
         }
 
-        public ApprenticeFeedbackDataContext(IOptions<ApplicationSettings> config, DbContextOptions<ApprenticeFeedbackDataContext> options, AzureServiceTokenProvider azureServiceTokenProvider) : base(options)
+        public ApprenticeFeedbackDataContext(IOptions<ApplicationSettings> config, DbContextOptions<ApprenticeFeedbackDataContext> options, ChainedTokenCredential azureServiceTokenProvider) : base(options)
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ApprenticeFeedback.Data/ApprenticeFeedbackDataContext.cs:63]: And rename the variable here

File: src/SFA.DAS.ApprenticeFeedback.Api/AppStart/AddDatabaseRegistrations.cs [hunk 3]
```diff
@@ -23,7 +23,12 @@ public static void AddDatabaseRegistration(this IServiceCollection services, ICo
             }
             else
             {
-                services.AddSingleton(new AzureServiceTokenProvider());
+                services.AddSingleton(new ChainedTokenCredential(
+                    new ManagedIdentityCredential(),
```
Comment (ChrisJWoodcock) [inline src/SFA.DAS.ApprenticeFeedback.Api/AppStart/AddDatabaseRegistrations.cs:27]: Did you do any investigation about what these additional singletons where needed for other than the ChainedTokenCredential?

**Review by ChrisJWoodcock — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ApprenticeFeedback.Api/AppStart/AddDatabaseRegistrations.cs (hunk 3 above)
Comment (SawNawDfE) [reply src/SFA.DAS.ApprenticeFeedback.Api/AppStart/AddDatabaseRegistrations.cs:27]: From what I understand, `ChainedTokenCredential` accepts any number of objects derived from `TokenCredential`, and tries each of them until authentication is successful. This is the combination that I've found to work, but others might work as well

**Pull request merged**